src/Appliance_Item.php

```diff
@@ -63,7 +63,7 @@ public static function getTypeName($nb = 0)
 
     public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
     {
-        if (!Appliance::canView()) {
+        if (!Appliance::canView() || !$item instanceof CommonDBTM) {
```

Comment:
Replacing `$item->getType()` by `::class` would also probably fix these issues without having to add this additional test. The same remark applies to many changes made in this PR.

`getType()` is not deprecated because we did not had time to change it everywhere, but the plan is to use `::class` instead of `getType()` everywhere we can.

Reply:
I've fixed a few getType in these changes, maybe I missed some.
I'll do another pass later.